RecoEgamma/ElectronIdentification/src/ElectronMVAEstimatorRun2Fall17.cc

```diff
@@ -0,0 +1,475 @@
+  rhoLabel_               (edm::InputTag                              ("fixedGridRhoFastjetAll")),
+  ptSplit_                (conf.getParameter<double>                  ("ptSplit")),
+  ebSplit_                (conf.getParameter<double>                  ("ebSplit")),
+  ebeeSplit_              (conf.getParameter<double>                  ("ebeeSplit")),
+  varNames_               (conf.getParameter<std::vector<std::string>>("varNames"))
+{
+
+  const std::vector <std::string> weightFileNames
+    = conf.getParameter<std::vector<std::string> >("weightFileNames");
+
+  std::vector<double> clipsLowerValues = conf.getParameter<std::vector<double>>("clipLower");
+  std::vector<double> clipsUpperValues = conf.getParameter<std::vector<double>>("clipUpper");
+
+  // Initialize GBRForests and and clipping instructions
+  init(weightFileNames);
+  setClips(clipsLowerValues, clipsUpperValues);
+
+  withIso_ = withIso;
+  debug_ = conf.getUntrackedParameter<bool>("debug", false);
+
+}
+
+ElectronMVAEstimatorRun2Fall17::ElectronMVAEstimatorRun2Fall17(
+        const std::string &mvaTag, const std::string &mvaName, bool withIso, const std::string &conversionsTag, const std::string &beamspotTag,
+        const double ptSplit, const double ebSplit, const double ebeeSplit, const bool debug):
+  AnyMVAEstimatorRun2Base( edm::ParameterSet() ),
+  tag_                    (mvaTag),
+  name_                   (mvaName),
+  methodName_             ("BDTG method"),
+  beamSpotLabel_          (edm::InputTag(beamspotTag)),
+  conversionsLabelAOD_    (edm::InputTag(conversionsTag)),
+  conversionsLabelMiniAOD_(conversionsLabelAOD_),
+  rhoLabel_               (edm::InputTag("fixedGridRhoFastjetAll")),
+  ptSplit_                (ptSplit),
+  ebSplit_                (ebSplit),
+  ebeeSplit_              (ebeeSplit)
+{
+
+  // Set if this is the ID with or without PF isolations
+  withIso_ = withIso;
+
+  // Set debug flag
+  debug_ = debug;
+
+}
+
+void ElectronMVAEstimatorRun2Fall17::init(const std::vector<std::string> &weightFileNames) {
+
+  // Initialize GBRForests 
+  if( (int)(weightFileNames.size()) != nCategories_ )
+    throw cms::Exception("MVA config failure: ")
+      << "wrong number of weightfiles" << std::endl;
+
+  gbrForests_.clear();
+  // Create a TMVA reader object for each category
+  for(int i=0; i<nCategories_; i++){
+
+    // Use unique_ptr so that all readers are properly cleaned up
+    // when the vector clear() is called in the destructor
+
+    edm::FileInPath weightFile( weightFileNames[i] );
+    gbrForests_.push_back( GBRForestTools::createGBRForest( weightFile ) );
+
+  }
+
+}
+
+void ElectronMVAEstimatorRun2Fall17::setClips(const std::vector<double> &clipsLowerValues, const std::vector<double> &clipsUpperValues) {
+
+  // Set up the variable clipping intructions
+  unsigned int i = 0;
+  for(auto const& value: clipsLowerValues) {
+      if (!isinf(value)) {
```

Comment:
Copying from the output of the static analyzer automatically run in the tests:
```
std::isnan / std::isinf does not work when fast-math is used. Please use edm::isNotFinite from 'FWCore/Utilities/interface/isNotFinite.h'
```
--